photon-api/src/main/scala/com/linkedin/photon/ml/data/RandomEffectDataSetPartitioner.scala

```diff
@@ -55,53 +56,50 @@ protected[ml] class RandomEffectDataSetPartitioner(idToPartitionMap: Broadcast[M
   }
 
   /**
+   * Hash code for this partitioner.
    *
-   * @return
+   * @return A Int hash code
    */
   override def hashCode: Int = idToPartitionMap.hashCode()
 
   /**
+   * For a given key, get the corresponding partition id. If the key is not in any partition, we randomly assign
+   * the training vector to a partition (with Spark's HashPartitioner).
    *
-   * @return
-   */
-  def numPartitions: Int = partitions
-
-  /**
-   *
-   * @param key
-   * @return
+   * @param key A training vector key (String).
+   * @return The partition id to which the training vector belongs.
    */
   def getPartition(key: Any): Int = key match {
     case string: String =>
-      idToPartitionMap.value.getOrElse(string, defaultPartitioner.getPartition(string))
+      idToPartitionMap.value.getOrElse(string, new HashPartitioner(numPartitions).getPartition(string))
     case any =>
       throw new IllegalArgumentException(s"Expected key of ${this.getClass} is String, but ${any.getClass} found")
   }
-
-  /**
-   *
-   * @return
-   */
-  def defaultPartitioner: HashPartitioner = new HashPartitioner(partitions)
 }
 
 object RandomEffectDataSetPartitioner {
   /**
-   * Generate a partitioner from the random effect dataset.
+   * Generate a partitioner for one type of random effect model.
```

Comment:
Had a discussion offline about it. Going back to original phrasing but we might have a task to sort out our field names...